README.md

```diff
@@ -1,75 +1,138 @@
+The Onload Operator v3 depends on third-party software:
 
-### Test It Out
-1. Install the CRDs into the cluster:
+1. Kernel Module Manamagent (KMM) Operator v1.1.1.
+2. CNI that supports ipvlan or macvlan configurations. The Onload Operator currently suggests Multus CNI. A sample configuration is https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/examples/macvlan-pod.yml.
 
-```sh
-make install
-```
 
-2. Run your controller (this will run in the foreground, so switch to a new terminal if you want to leave it running):
+### Build and deploy Onload Operator v3
 
-```sh
-make run
+Please install Go 1.21+ in the machines where you run `make` and follow these instructions in this repository:
+
+1. Create and push the Onload Operator controller image:
+```
+make docker-build docker-push IMG=kitchen-sink.kube.test:5000/operator:latest
+```
+2. Create and push the Onload Device Plugin image:
 ```
+make device-plugin-docker-build device-plugin-docker-push DEVICE_IMG=kitchen-sink.kube.test:5000/deviceplugin:latest
+```
+3. Deploy the Onload Operator v3:
+```
+make deploy IMG=kitchen-sink.kube.test:5000/operator:latest
+```
+4. Patch the Onload CR sample accordingly:
+```diff
+diff --git a/config/samples/onload_v1alpha1_onload.yaml b/config/samples/onload_v1alpha1_onload.yaml
+index 0ffba0c..2768bdc 100644
+--- a/config/samples/onload_v1alpha1_onload.yaml
++++ b/config/samples/onload_v1alpha1_onload.yaml
+@@ -55,9 +55,9 @@ spec:
+     # Example image locations using openshift local image registry.
+     kernelMappings:
+       - regexp: '^.*\.x86_64$'
+-        kernelModuleImage: image-registry.openshift-image-registry.svc:5000/onload-clusterlocal/onload-module:v8.1.0-${KERNEL_FULL_VERSION}
++        kernelModuleImage: kitchen-sink.kube.test:5000/onload-module:deccdb8d036a4c7794dea2cda106b3c112b374a9-4.18.0-372.49.1.el8_6.x86_64
+         sfc: {}
+-    userImage: image-registry.openshift-image-registry.svc:5000/onload-clusterlocal/onload-user:v8.1.0
++    userImage: kitchen-sink.kube.test:5000/onload-user:latest
+     version: 8.1.0
+     imagePullPolicy: Always
+   devicePlugin:
+```
+Please also make sure `devicePluginImage` is correct. Another important field is the node `selector`, which tells the Onload Operator where to deploy the Onload Device Plugin DaemonSet and Module kind (KMM).
 
-**NOTE:** You can also run this in one step by running: `make install run`
+5. Finally, deploy the Onload CR:
+```
+oc apply -k config/samples/
+```
 
-### Modifying the API definitions
-If you are editing the API definitions, generate the manifests such as CRs or CRDs using:
 
-```sh
-make manifests
+## Run Onloaded application
+
+At this point, Onload is deployed at the cluster, and the users can run Onloaded applications, e.g.
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test
+  annotations:
+    k8s.v1.cni.cncf.io/networks: ipvlan-sf0
```

Comment:
Could you add a note/comment for this line saying that this should be the name of the NetworkAttachmentDefinition that they created earlier (and perhaps make it obvious that they need to create the NetworkAttachmentDefinition, just having the CNI isn't sufficient).